SudokuSolver/SudokuSolver/CrossHatching/SudokuSolver.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Microsoft.Toolkit.Extensions;
+using SudokuSolver.Common;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SudokuSolver.CrossHatching
+{
+    class SudokuSolver : BaseSudokuSolver
+    {
+        private Dictionary<int, List<(int i, int j)>> filledPos;
+        private Dictionary<int, int> remaining;
+        private Dictionary<int, Dictionary<int, List<int>>> graph;
+
+        public SudokuSolver() { }
+        public override Result<Sudoku> Solve(Sudoku sudoku)
+        {
+            BuildFilledPosAndRemaining(sudoku);
+            BuildGraph(sudoku);
+
+            remaining = remaining.OrderBy(kvpair => kvpair.Value).ToDictionary(x => x.Key, x => x.Value);
+
+            var remainingKeys = remaining.Keys.ToList();
+            int currentRemKey = 0;
+            int currentRow = 0;
+
+            Stopwatch sw = new Stopwatch();
+            sw.Start();
+            var result = TrySolve(
+                sudoku,
+                currentRemKey,
+                remainingKeys,
+                currentRow,
+                graph[remainingKeys[currentRemKey]].Keys.ToList()
+            );
+            sw.Stop();
+            Console.WriteLine(sw.Elapsed);
+
+            return result;
+        }
+
+        private Result<Sudoku> TrySolve(Sudoku sudoku, int currentRemKey, List<int> remainingKeys, int currentRow, List<int> rows)
+        {
+            var k = remainingKeys[currentRemKey];
```

Comment:
trochę dużo mi w tej funkcji jednoliterówek, przez co trochę ciężko się to czyta